### PR TITLE
MULE-18030: HttpPolicyPointcutParameters, only use the headers if a

### PR DESCRIPTION
**Pull request by elrodro83**
pointcut actually needs them

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/policy/PolicyProvider.java [hunk 1]
```diff
@@ -29,6 +29,22 @@
    */
   List<Policy> findSourceParameterizedPolicies(PolicyPointcutParameters policyPointcutParameters);
 
+  /**
+   * @return {@code true} if there is at least one source policy applied that relies on the headers to check for the pointcut
+   *         match.
+   *
+   * @since 4.3
+   */
+  boolean isAnySourcePolicyHeadersAware();
```
Comment (balbifm) [inline core/src/main/java/org/mule/runtime/core/api/policy/PolicyProvider.java]: This is coupling HTTP to the policy engine. It should probably be in http-policy-api

File: core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyManager.java [hunk 2]
```diff
@@ -200,6 +200,11 @@ public SourcePolicy createSourcePolicyInstance(Component source, CoreEvent sourc
     return sourcePolicy;
   }
 
+  @Override
+  public boolean isAnySourcePolicyHeadersAware() {
```
Comment (balbifm) [inline core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyManager.java]: I got lost, where is this used?

Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyManager.java]: removed this

File: core/src/main/java/org/mule/runtime/core/api/policy/PolicyProvider.java (hunk 1 above)
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/api/policy/PolicyProvider.java]: yep, fixed

**Pull request merged**